app/controllers/session_controller.rb

```diff
@@ -408,12 +405,12 @@ def forgot_password
     RateLimiter.new(nil, "forgot-password-hr-#{request.remote_ip}", 6, 1.hour).performed!
     RateLimiter.new(nil, "forgot-password-min-#{request.remote_ip}", 3, 1.minute).performed!
 
-    user = User.find_by_username_or_email(params[:login])
+    user = User.find_by_username_or_email(normalized_login_param)
 
     if user
       RateLimiter.new(nil, "forgot-password-login-day-#{user.username}", 6, 1.day).performed!
     else
-      RateLimiter.new(nil, "forgot-password-login-hour-#{params[:login].to_s[0..100]}", 5, 1.hour).performed!
+      RateLimiter.new(nil, "forgot-password-login-hour-#{normalized_login_param}", 5, 1.hour).performed!
```

Comment:
The only thing I can say here is that if the normalised login param ends up nil, this rate limit is kind of useless, because the key will be `forgot-password-login-hour-`. Maybe we need `if normalized_login_param.present?` like the 2FA route?

Reply:
true ... I guess it is not too terrible though cause a rate limit on people somehow bypass the required param with a bunch of unicode spaces it is not the end of the world if we rate limit them.